apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -36,9 +41,40 @@ import { createDefaultPage } from "./page.service"
 const ajv = new Ajv({ allErrors: true, strict: false, logger: false })
 const schemaValidator = ajv.compile<IsomerSchema>(schema)
 
+const validateUserPermissions = async ({
```

Comment:
from karrui: we can consider consolidating all the resources that the user has access to, and then set the ability itself on the context. 

the benefit is that the callsite is simpler + the logic is consolidated into one place